src/shared/ListRepo/ReposTableTeam/ReposTableTeam.tsx

```diff
@@ -0,0 +1,274 @@
+  sorting: Array<{ id: string; desc: boolean }>
+) {
+  const state = sorting.at(0)
+
+  if (state) {
+    const direction = state.desc
+      ? OrderingDirection.DESC
+      : OrderingDirection.ASC
+
+    let ordering
+    if (state.id === 'name') {
+      ordering = TeamOrdering.NAME
+    }
+
+    if (state.id === 'latestCommitAt') {
+      ordering = TeamOrdering.COMMIT_DATE
+    }
+
+    return { direction, ordering }
+  }
+
+  return undefined
+}
+
+const columnHelper = createColumnHelper<Repository>()
+
+interface ReposTableTeamProps {
+  searchValue: string
+}
+
+const getColumns = ({
+  inactive,
+  isCurrentUserPartOfOrg,
+}: {
+  inactive: boolean
+  isCurrentUserPartOfOrg: boolean
+}) => {
+  const nameColumn = columnHelper.accessor('name', {
+    header: 'Name',
+    id: 'name',
+    cell: (info) => {
+      const repo = info.row.original
+      return (
+        <RepoTitleLink
+          repo={repo}
+          showRepoOwner={!!repo?.author?.username}
+          pageName={!!repo?.active ? 'repo' : 'new'}
+          disabledLink={!isCurrentUserPartOfOrg && !repo?.active}
+        />
+      )
+    },
+  })
+
+  if (inactive) {
+    return [
+      nameColumn,
+      columnHelper.accessor('lines', {
+        header: '',
+        id: 'lines',
+        cell: (info) => {
+          const repo = info.row.original
+          return (
+            <InactiveRepo
+              owner={repo?.author?.username ?? ''}
+              repoName={repo?.name}
+              isCurrentUserPartOfOrg={isCurrentUserPartOfOrg}
+              isActive={!!repo?.active}
+            />
+          )
+        },
+      }),
+    ]
+  }
+  return [
+    nameColumn,
+    columnHelper.accessor('latestCommitAt', {
+      header: 'Last updated',
+      id: 'latestCommitAt',
+      cell: (info) => {
+        return (
+          <span className="text-ds-gray-quinary">
+            {info?.renderValue() ? formatTimeToNow(info?.renderValue()) : ''}
+          </span>
+        )
+      },
+    }),
+    columnHelper.accessor('lines', {
+      header: 'Tracked lines',
+      id: 'lines',
+      cell: (info) => {
+        const repo = info.row.original
+        return typeof repo?.lines === 'number' && !!repo?.active ? (
+          <span>{repo.lines}</span>
+        ) : (
+          <InactiveRepo
+            owner={repo?.author?.username ?? ''}
+            repoName={repo?.name}
+            isCurrentUserPartOfOrg={isCurrentUserPartOfOrg}
+            isActive={!!repo?.active}
+          />
+        )
+      },
+    }),
+  ]
+}
+
+const ReposTableTeam = ({ searchValue }: ReposTableTeamProps) => {
+  const [sorting, setSorting] = useState<SortingState>([
+    {
+      id: 'latestCommitAt',
+      desc: true,
+    },
+  ])
+  const { owner } = useParams<{ owner: string }>()
+
+  const repoDisplay = useContext(ActiveContext)
+  const activated =
+    repoDisplayOptions[
+      repoDisplay.toUpperCase() as keyof typeof repoDisplayOptions
+    ]?.status
```

Comment:
Can you create a ticket noted down to refactor this context to Typescript and then remove this casting here, just so it's tracked and we can remove it down the road. As well as noting this specific code section here in the ticket.

You can link this [doc here](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/context/#basic-example) for doing context in a typesafe manner.

Reply:
https://github.com/codecov/gazebo/issues/2379